root/static/scripts/edit/URLCleanup.js

```diff
@@ -4300,6 +4303,36 @@ const CLEANUPS: CleanupEntries = {
       return url;
     },
   },
+  'vimeoondemand': {
+    match: [new RegExp(
+      '^(https?://)?([^/]+\\.)?vimeo\\.com/(?:ondemand|store/ondemand)',
+      'i',
+    )],
+    restrict: [LINK_TYPES.downloadpurchase, LINK_TYPES.streamingpaid],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?vimeo\.com\/ondemand\/([^\/?#]+)(?:.*)$/, 'https://vimeo.com/ondemand/$1');
+      return url;
+    },
+    validate: function (url) {
+      const isStoreLink = /^https:\/\/vimeo\.com\/store\/ondemand/.test(url);
```

Comment:
Because `/store/ondemand` links are not modified by the function `clean`, the function `validate` should catch unclean URLs too:
```suggestion
      const isStoreLink = /vimeo\.com\/store\/ondemand/.test(url);
```